### ISSUE TITLE
Integration and validation tests with b=0 only

### ISSUE THREAD

**Issue by oesteban** [nipreps/eddymotion#40]
Two low-hanging fruit tests using only a b=0:

- Validation:
    - [x] Unit tests of the `TrivialB0Model` implementation
    - [x] Unit tests of ANTs' config for *b=0*.
- Integration:
    - [x] Introduce a series of transforms (e.g., copy them from another dataset) and resample the *b=0* through that motion series. Then run the whole estimator and check the proximity of the estimated transforms.

**Comment by oesteban** [comment]
Okay, I've tried to set up some basics, and implemented a unit-test for the `TrivialB0Model` sketching what we intend to do here:

https://github.com/nipreps/eddymotion/blob/152e0d4aeb6917965ed1821ea1507fb68630fa3b/eddymotion/tests/test_model.py#L29-L42

> Unit tests of ANTs' config for b=0.

I see this implemented in [`test_estimator.py`](https://github.com/nipreps/eddymotion/blob/main/eddymotion/tests/test_estimator.py). To do so:

- [ ] Upload an example B0 map (3D)
- [ ] Apply random (and small) rotations and translations on the image. These can be generated with `nibabel.eulerangles.euler2mat` (e.g., https://github.com/poldracklab/nitransforms/blob/master/nitransforms/tests/test_linear.py#L107). For the resampling, I would use `nitransforms.linear.Affine.resample()`.
- [ ] Run the registration interface loading the parameters from config
- [ ] Check that the error of the affine is below some threshold when transforming points scattered across the volume.

>  Introduce a series of transforms (e.g., copy them from another dataset) and resample the b=0 through that motion series. Then run the whole estimator and check the proximity of the estimated transforms.

I see this implemented in [`test_integration.py`](https://github.com/nipreps/eddymotion/blob/main/eddymotion/tests/test_integration.py). To do so:
- [ ] Upload an example B0 map (3D) - or reuse from the previous test
- [ ] Generate a series of moved B0's with the realignment matrices from another, real dataset.
- [ ] Initiate the `TrivialB0Model` with this B0 map and feed the dwi object with the moved data (simulate perhaps several transforms).

**Comment by oesteban** [comment]
Hi there! I've uploaded some data under https://github.com/nipreps/eddymotion/tree/main/eddymotion/tests/data, generated by @dPys with FiberFox.

**Comment by sebastientourbier** [comment]
@oesteban I am progressing! For now, I managed to:
- extract the b0 volume from the `sub-01_dwi.nii.gz` provided by Céline.
- run AFNI 3dvolreg on the fmri of the first run, first subject of ds000005 (240 time frames) on openneuro with the middle frame as reference. This gives me a text file where one line corresponds to the motion parameters for one time frame.
- make 20 copies of the b0 where I applied the motion parameters from the last 20 frames (with nitransforms), saved in temporary niftis.
- save a unique 4D nifti image that concatenate the original and the moving b0s

I haven't opened a PR yet but here is a quick preview of the motion introduced

![b0 moving](https://user-images.githubusercontent.com/22279770/141146465-df5f7c12-0bf6-4787-a0b3-ee92cc5c0c55.gif)

What do think?
